### ISSUE TITLE
cjxl GIF to JXL crashes using VarDCT with effort-8 or effort-9

### ISSUE THREAD

**Issue by Traneptora** [libjxl/libjxl#1929]
**Describe the bug**
Consider the following GIF: [e4ec7c5d7af5342f57347c9ada429fba.gif](https://user-images.githubusercontent.com/1342577/204566989-edf60687-cce2-4f5b-a2d8-4d7e5d8579f1.gif)
![e4ec7c5d7af5342f57347c9ada429fba](https://user-images.githubusercontent.com/1342577/204566989-edf60687-cce2-4f5b-a2d8-4d7e5d8579f1.gif)

**To Reproduce**
```
cjxl -d 1 -e 8 e4ec7c5d7af5342f57347c9ada429fba.gif output.jxl
```

**Expected behavior**
cjxl should produce a JXL file as it does with `-e 7` and lower.

**Actual behavior**
Using `-e 8` or `-e 9` crashes. Using `-e 7` and lower does not crash.
```
$ cjxl -e 8 -d 1 e4ec7c5d7af5342f57347c9ada429fba.gif output.jxl
JPEG XL encoder v0.8.0 759a29f9 [AVX2]
./lib/extras/dec/color_hints.cc:54: No color_space/icc_pathname given, assuming sRGB
Read 600x563 image, 3032897 bytes, 3.8 MP/s
Encoding [VarDCT, d1.000, effort: 8], 
./lib/jxl/image.h:57: JXL_CHECK: xsize <= orig_xsize_
Illegal instruction (core dumped)
```
Valgrind output: [valgrind.log](https://github.com/libjxl/libjxl/files/10114328/valgrind.log)

**Environment**
 - OS: Arch Linux x86_64
 - Compiler version: gcc 12.2.0
 - cjxl/djxl version string: JPEG XL encoder v0.8.0 759a29f9 [AVX2]

**Additional context**
Here, `-d 1` is just to force VarDCT. If you don't include that it implicitly defaults to `-d 0` and uses modular lossless.

**Comment by mo271** [comment]
Thanks for the report! 
Bisecting this shows that this used to work before we started using the render pipeline in the encoder: https://github.com/libjxl/libjxl/commit/60654d7e75829fa94e557e6cf3f1d47d5456fbe5.

**Comment by AlifianK** [comment]
Fixed in #2556?

**Comment by Traneptora** [comment]
Appears to be fixed :+1:
